st7920.py

```diff
@@ -127,8 +129,20 @@ def put_text(self, s, x, y):
 			except KeyError:
 				pass
 			x += cw
-	
-	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63):
-		for i in range(dy1, dy2+1):
-			self.send(0,0,[0x80 + i%32, 0x80 + ((dx1/16) + (8 if i>=32 else 0))]) # set address
-			self.send(1,0,self.fbuff[i][dx1/8:(dx2/8)+1])
+
+	def _send_line(self, row, dx1, dx2):
+		self.send(0,0,[0x80 + row%32, 0x80 + ((dx1//16) + (8 if row>=32 else 0))]) # set address
+		self.send(1,0,self.fbuff[row][dx1//8:(dx2//8)+1])
+
+	def redraw(self, dx1=0, dy1=0, dx2=127, dy2=63, full=False):
+		if self.currentlydisplayedfbuff == None: # first redraw always affects the complete LCD
+			for row in range(0, 64):
+				self._send_line(row, dx1, dx2)

```

Comment:
This needs to be `self._send_line(row, 0, 127)`, so the whole row is sent to the LCD the first time.